### PR TITLE
Fix `plugins` and `overrides` duplicates when merged 

### PR DESCRIPTION
**Pull request by kuoruan**
We merge a.plugins and b.plugins, but only remove duplicates when b.plugins exist. Duplicate items can be in a.plugins too.

> Which issue, if any, is this issue related to?

None.

> Is there anything in the PR that needs further explanation?

No, it's self-explanatory.

### CODE REVIEW

File: lib/augmentConfig.js [hunk 1]
```diff
@@ -250,8 +252,10 @@ function mergeConfigs(a, b) {
 		}
 
 		if (b.overrides) {
-			overridesMerger.overrides = [...new Set(overridesMerger.overrides.concat(b.overrides))];
+			overridesMerger.overrides = overridesMerger.overrides.concat(b.overrides);
```
Comment (Mouvedia) [inline lib/augmentConfig.js:255]: question

Why didn't you do it on l251 as well?

File: lib/augmentConfig.js [hunk 2]
```diff
@@ -250,8 +252,10 @@ function mergeConfigs(a, b) {
...
 		}
+
+		overridesMerger.overrides = [...new Set(overridesMerger.overrides)];
```
Comment (Mouvedia) [inline lib/augmentConfig.js:258]: If you are doing it here, you don't need to it before

File: lib/augmentConfig.js [hunk 3]
```diff
@@ -235,8 +235,10 @@ function mergeConfigs(a, b) {
 		}
 
 		if (b.plugins) {
-			pluginMerger.plugins = [...new Set(pluginMerger.plugins.concat(b.plugins))];
+			pluginMerger.plugins = pluginMerger.plugins.concat(b.plugins);
 		}
+
+		pluginMerger.plugins = [...new Set(pluginMerger.plugins)];
```
Comment (Mouvedia) [inline lib/augmentConfig.js:241]: ditto

Comment (ybiquitous) [reply lib/augmentConfig.js:241]: I also agree with @Mouvedia. We can pass `lib/__tests__/applyOverrides.test.js` even if using `lib/augmentConfig.js` before this change.

**Comment by kuoruan** [comment]
@Mouvedia @ybiquitous Updated, please review again.

File: lib/augmentConfig.js [hunk 4]
```diff
@@ -354,7 +361,8 @@ function applyOverrides(fullConfig, rootConfigDir, filePath) {
 	let { overrides, ...config } = fullConfig;
 
 	if (!overrides) {
-		return config;
+		// Do mergeConfigs to remove duplicate items
+		return mergeConfigs(config, {});
```
Comment (Mouvedia) [inline lib/augmentConfig.js:365]: opinion

that's overkill
i.e. if someone has only one config, which has duplicates, he ought to fix them himself

File: lib/augmentConfig.js [hunk 5]
```diff
@@ -250,8 +252,13 @@ function mergeConfigs(a, b) {
 		}
 
 		if (b.overrides) {
-			overridesMerger.overrides = [...new Set(overridesMerger.overrides.concat(b.overrides))];
+			overridesMerger.overrides = overridesMerger.overrides.concat(b.overrides);
 		}
+
+		// Remove duplicates, last one wins
+		overridesMerger.overrides = overridesMerger.overrides.filter(
```
Comment (Mouvedia) [inline lib/augmentConfig.js:259]: question

Overrides are objects, where are you doing the deep comparison to detect duplicates?

**Comment by kuoruan** [comment]
Seems no need for this PR, closing it.

**Pull request closed without merging**